### PR TITLE
Add Auth Middleware

### PR DESCRIPTION
**Pull request by magaofei**
### What was a problem?

```
Using middleware is more unified and standardized
Django has a limit on the size of the file upload, increasing the size
```

### How this PR fixes the problem?

```
DESCRIBE HERE
```

### Check lists (check `x` in `[ ]` of list items)

- [x] Run MobSF unit tests
- [ ] Tested Working on Linux, Mac, and Windows
- [x] Coding style (indentation, etc)

### Additional Comments (if any)

```
DESCRIBE HERE
```

### CODE REVIEW

File: MobSF/settings.py [hunk 1]
```diff
@@ -225,6 +229,9 @@
 # https://docs.djangoproject.com/en/dev/howto/static-files/
 STATIC_URL = '/static/'
 
+# 500MB
+DATA_UPLOAD_MAX_MEMORY_SIZE = 524288000
```
Comment (ajinabraham) [inline MobSF/settings.py]: Can you keep this to `256 MB` as we have that followed elsewhere.

Comment (magaofei) [reply MobSF/settings.py]: Should I change it to 256MB or delete it?

Comment (ajinabraham) [reply MobSF/settings.py]: Please change that to equivalent of 256 MB

File: MobSF/rest_api.py [hunk 2]
```diff
@@ -41,136 +41,119 @@ def api_auth(meta):
...
-                response = make_api_response(resp, 500)
-            else:
-                response = make_api_response(resp)
+    if request.method == 'POST':
+        resp = upload(request, True)
+        if "error" in resp:
+            response = make_api_response(resp, 500)
         else:
-            response = make_api_response({"error": "Method Not Allowed"}, 405)
+            response = make_api_response(resp)
```
[the remark this answers is not in the document]
Comment (magaofei) [reply MobSF/rest_api.py]: If it is a GET request, it will return 405, default is 200.

**Comment by magaofei** [comment]
```
[INFO] ALL TESTS COMPLETED!
[INFO] Test Status: all tests completed
[21/Aug/2018 03:10:49] "GET /runapitest/ HTTP/1.1" 200 40
```

**Comment by magaofei** [comment]
Thank you. Do you have time to look at #688 ? This is really a problem for some apk.

**Pull request merged**